proto/aserto/directory/common/v3/common.proto

```diff
@@ -7,84 +7,309 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/common/
 import "google/api/field_behavior.proto";
 import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
+import "buf/validate/validate.proto";
+
 
 message Object {
-    // external object identifier (cs-string)
-    string id                                           = 1     [(google.api.field_behavior) = REQUIRED];
     // object type name
-    string type                                         = 2     [(google.api.field_behavior) = REQUIRED];
+    string type                                         = 1
+    [
+        (buf.validate.field) = {
+            required: true,
+            cel: {
+                id: "object.type"
+                message: "must be all lowercase, start with a letter, can contain letters, digits, underscores, and dashes, and must end with a letter or digit"
+                expression: "this.matches('^[a-z][a-z0-9_-]{1,62}[a-z0-9]$')"
+            }
+            string: {
+                max_len: 64
+            }
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+    // external object identifier (cs-string, no spaces or tabs)
+    string id                                           = 2
+    [
+        (buf.validate.field) = {
+            required: true,
+            cel: {
+                id: "object.id"
+                message: "cannot contain any spaces or other whitespace characters"
+                expression: "this.matches('^[^\\t\\n\\f\\r ]+$')"
```

Comment:
You should be able to use `\S` to match non-whitespace characters:
`this.matches('^\\S+$')`

Reply:
I know, that triggers a bug in protovalidate.

Reply:
@ronenh see https://bufbuild.slack.com/archives/CRZ680FUH/p1690577643575079 and https://github.com/gertd/grpc-test

Reply:
Double escaping provided the workaround, updated the PR accordingly.
Still unclear why `'^[\\S]+$'` fails and `'^[^\\t\\n\\f\\r ]+$'` succeeds